hrjobcontract/tests/phpunit/CRM/Hrjobcontract/BAO/HRJobDetailsTest.php

```diff
@@ -18,61 +18,77 @@ public function setUpHeadless() {
-      ['period_start_date' => '2016-01-03', 'period_end_date'=> '2016-01-05'],
-      ['period_start_date' => '2016-01-01', 'period_end_date'=> '2016-01-02'],
-      ['period_start_date' => '2016-01-31', 'period_end_date'=> '2016-02-01'],
-      ['period_start_date' => '2016-02-10', 'period_end_date'=> '2016-02-20'],
-      ['period_start_date' => '2015-01-01'],
-      ['period_start_date' => '2016-03-01'],
-    ];
-
-    foreach ($sampleFailedDates as $datePair)  {
-      $params = ['contact_id' => $contactID, 'period_start_date' => $datePair['period_start_date']];
-      if (!empty($datePair['period_end_date']))  {
-        $params['period_end_date'] = $datePair['period_end_date'];
-      }
-
-      $result = CRM_Hrjobcontract_BAO_HRJobDetails::validateDates($params);
 
-      $this->assertFalse($result['success']);
+    $params = ['contact_id' => $contactID, 'period_start_date' => $start_date];
+    if (!empty($end_date))  {
+      $params['period_end_date'] = $end_date;
     }
 
-    // Now we test against valid dates so we expect the result of
-    // 'validateDates' call to be TRUE.
+    $result = CRM_Hrjobcontract_BAO_HRJobDetails::validateDates($params);
 
-    $sampleCorrectDates = [
-      ['period_start_date' => '2015-01-01', 'period_end_date'=> '2015-05-31'],
-      ['period_start_date' => '2016-01-21', 'period_end_date'=> '2016-01-31'],
-    ];
+    $this->assertFalse($result['success']);
+  }
+
+  /**
+   * @param $start_date
+   * @param $end_date
+   *
+   * @dataProvider validDatesProvider
+   */
+  public function testValidateDatesWithValidDates($start_date, $end_date) {
 
-    foreach ($sampleCorrectDates as $datePair)  {
-      $params = ['contact_id' => $contactID, 'period_start_date' => $datePair['period_start_date']];
-      if (!empty($datePair['period_end_date']))  {
-        $params['period_end_date'] = $datePair['period_end_date'];
-      }
+    // Create test Contact.
+    $contactParams = array("first_name" => "chrollo", "last_name" => "lucilfer");
+    $contactID =  $this->createContact($contactParams);
 
-      $result = CRM_Hrjobcontract_BAO_HRJobDetails::validateDates($params);
+    // Create Two test Job Contracts.
+    $this->createJobContract($contactID, '2016-01-01', '2016-01-10');
+    $this->createJobContract($contactID, '2016-02-01');
 
-      $this->assertTrue($result['success']);
+    $params = ['contact_id' => $contactID, 'period_start_date' => $start_date];
+    if (!empty($end_date))  {
+      $params['period_end_date'] = $end_date;
     }
+
+    $result = CRM_Hrjobcontract_BAO_HRJobDetails::validateDates($params);
+
+    $this->assertTrue($result['success']);
+  }
+
+  public function invalidDatesProvider()
+  {
+    return [
+      ['2016-01-01', '2016-10-10'],
+      ['2016-01-03', '2016-01-05'],
+      ['2016-01-01', '2016-01-02'],
+      ['2016-01-31', '2016-02-01'],
+      ['2016-02-10', '2016-02-20'],
+      ['2015-01-01'],
+      ['2016-03-01'],
```

Comment:
The testValidateDatesWithInvalidDates is failing because of the missing values for end_date in the last two itens of the data provider.  You can fix it by passing `null` as the end_date value:
![captura de tela 2016-08-15 as 15 46 43](https://cloud.githubusercontent.com/assets/388373/17675350/cc10d422-62ff-11e6-9c43-65ef197203c4.png)